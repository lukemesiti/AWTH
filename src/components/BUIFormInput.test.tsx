import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { BUIFormInput, FormElement } from ".";

describe("BUIFormInput", () => {
  it("should render with element values", () => {
    // Arrange
    const element: FormElement = {
      label: "Email",
      type: "text",
      value: "",
    };

    // Act
    const result = render(<BUIFormInput element={element} />);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
