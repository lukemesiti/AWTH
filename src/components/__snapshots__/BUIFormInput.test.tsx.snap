// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BUIFormInput > should render with element values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label>
        Email
      </label>
      <input
        class="block w-full rounded-xl border-0 p-1.5 text-gray-900 ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-green-600 md:text-lg sm:text-md sm:leading-6"
        type="text"
        value=""
      />
    </div>
  </body>,
  "container": <div>
    <label>
      Email
    </label>
    <input
      class="block w-full rounded-xl border-0 p-1.5 text-gray-900 ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-green-600 md:text-lg sm:text-md sm:leading-6"
      type="text"
      value=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
